waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -339,6 +340,16 @@
   ;; see https://clojure.org/reference/reader#_literals
   (keyword (str "p" port-number)))
 
+(defn retrieve-failing-container-names
+  "Retrieves the names of containers that are failing (i.e. in a CrashLoopBackOff)."
+  [container-statuses type-filter]
+  (->> container-statuses
+    (filter (fn retrieve-failing-containers-crash-loop? [{:keys [ready reason type]}]
+              (and (= type type-filter)
+                   (false? ready)
+                   (= "CrashLoopBackOff" reason))))
+    (map :name)))
```

Comment:
This would be simpler as a `for` expression.

Reply:
Changed to `for`.